libs/mail/api/src/lib/mail-provider/mailchimp-mail-provider.ts

```diff
@@ -52,17 +52,29 @@ function flattenObjForMandrill<T>(ob: T): Record<string, string> {
   for (const i in ob) {
     const nestedObj = ob[i]
 
-    if (nestedObj && typeof nestedObj === 'object' && !Array.isArray(nestedObj)) {
+    if (Array.isArray(nestedObj)) {
+      for (const j in nestedObj) {
+        if (nestedObj[j] && typeof nestedObj[j] === 'object') {
```

Comment:
Added this as this fixes a bug of the other PR with `null`, `number` `string` arrays (or rather any non nested arrays or arrays of objects).